core/src/main/java/com/amplifyframework/core/plugin/Plugin.java

```diff
@@ -40,9 +39,9 @@
      * Configure the plugin with customized configuration object.
      * @param pluginConfiguration plugin-specific configuration data
      * @param context An Android Context
-     * @throws AmplifyException when configuration for a plugin was not found
+     * @throws AmplifyException an error is encountered during configuration.
      */
-    void configure(@NonNull JSONObject pluginConfiguration, Context context) throws AmplifyException;
+    void configure(JSONObject pluginConfiguration, Context context) throws AmplifyException;
```

Comment:
Use `@Nullable` to indicate that the config may be `null` and the developer should check for that.

Reply:
This will be overrided with either `@NonNull` or `@Nullable` depending on the plugin, so I decided to leave it without.